packages/uni_ui/lib/restaurant_card.dart

```diff
@@ -0,0 +1,136 @@
+import 'package:flutter/material.dart';
+import 'package:uni_ui/generic_card.dart';
+import 'package:phosphor_flutter/phosphor_flutter.dart';
+
+class MenuItem {
+  final String name;
+  final String type;
+
+  MenuItem({required this.name, required this.type});
+}
+
+class RestaurantCard extends StatefulWidget {
+  final String title;
+  final String establishmentType;
+  final List<MenuItem> menuItems;
+
+  const RestaurantCard({
+    super.key,
+    required this.title,
+    required this.establishmentType,
+    required this.menuItems,
+  });
+
+  @override
+  _RestaurantCardState createState() => _RestaurantCardState();
+}
+
+class _RestaurantCardState extends State<RestaurantCard> {
+  bool isFavorite = false;
+
+  void _toggleFavorite() {
+    setState(() {
+      isFavorite = !isFavorite;
+    });
+  }
+
+  IconData? _getIconForMenuItemType(String type) {
```

Comment:
Extend a `StatelessWidget` instead.
The `isFavorite` state should be outside this widget and passed as a simple boolean argument. Also add the possibility to pass void function to run on favorite icon click.

Reply:
Okok! I'll fix that 😊